src/main/java/me/kixstar/dungeons/dungeons/ChestUtils.java

```diff
@@ -26,8 +38,15 @@ static void distributeItems(List<Chest> chests, List<ItemStack> items) {
      * @param items
      */
     static void distributeItems(Inventory inventory, List<ItemStack> items) {
-        // get random number between 0-26
-        random.nextInt(27);
+        if (inventory != null && items != null && !(items.isEmpty())) {
+            for (ItemStack item : items) {
+                int ramdomSlot = random.nextInt(27);
+                if (inventory.getItem(ramdomSlot) != null) {
```

Comment:
While the idea is great, it fails to distribute all items if the random number generator returns a slot number that's already gotten an item, as you only make one attempt at actually populating it, meaning there's potential for loss of items that should be distributed.